os/arch/arm/src/common/up_allocateheap.c

```diff
@@ -216,8 +216,9 @@ void up_addregion(void)
 	char *mem_size = CONFIG_HEAPx_SIZE;
 
 	for (region_cnt = 0; region_cnt < CONFIG_MM_REGIONS - 1; region_cnt++) {
-		if (*mem_start || *mem_size) {
-			dbg("CONFIG_HEAPx_BASE and CONFIG_HEAPx_SIZE are not defined properly\n");
+		if (!*mem_start || !*mem_size) {
+			dbg("Fail to add %dth heap region\n", region_cnt + 1);
+			break;
 		}
```

Comment:
If checked outside loop, code will be more readable and avoids abrupt exit from loop.

Reply:
mem_start and mem_size are updated at every loop. This is a conditional to check valid data. It can not be at outside of loop.

Reply:
OK, understood after viewed the whole function